ProcessSlave.php

```diff
@@ -474,24 +529,38 @@ protected function setupResponseLogging(\React\Http\Request $request, HttpRespon
                 '$http_referer',
                 '$http_user_agent',
             ], [
-                $_SERVER['REMOTE_ADDR'],
+                $remoteIp,
                 '-', //todo remote_user
                 $timeLocal,
                 $requestString,
                 $statusCode,
-                $response->getBytesSent(),
-                isset($request->getHeaders()['Referer']) ? $request->getHeaders()['Referer'] : '-',
-                isset($request->getHeaders()['User-Agent']) ? $request->getHeaders()['User-Agent'] : '-',
+                $size,
+                $request->hasHeader('Referer') ? $request->getHeaderLine('Referer') : '-',
+                $request->hasHeader('User-Agent') ? $request->getHeaderLine('User-Agent') : '-'
             ],
                 $this->logFormat);
 
             if ($response->getStatusCode() >= 400) {
                 $message = "<error>$message</error>";
             }
 
-
             $this->sendMessage($this->controller, 'log', ['message' => $message]);
-        });
+        };
+
+        if ($response->getBody() instanceof EventEmitterInterface) {
```

Comment:
This part looks ugly. Should we- just for the purpose of logging- really read the entire response into memory? Probably not a good idea, do we really need this?